pennylane/qnn/torch.py

```diff
@@ -295,6 +339,24 @@ def _evaluate_qnode(self, x):
                     qnode = functools.partial(qnode, x)
         return qnode().type(x.dtype)
 
+    def _evaluate_qnode_tape_mode(self, x):
+        """Evaluates a tape-mode QNode for a single input datapoint.
+
+        Args:
+            x (tensor): the datapoint
+
+        Returns:
+            tensor: output datapoint
+        """
+        kwargs = {**{self.input_arg: x}, **self.qnode_weights}
+        res = self.qnode(**kwargs).type(x.dtype)
+
+        if res.dim() == 0:
+            # result is 0-dimensional; add an extra dimension as the tests expect it
+            return torch.unsqueeze(res, dim=0)
```

Comment:
Oh, I actually know why this occurs. We need to modify `pennylane.tape.qnode` to incorporate this behaviour, rather then doing this here. I think this test suite just caught this differing behaviour in tape mode for the first time.

Reply:
I have fixed the QNode, so that this can be removed :)